src/ucp/wireup/wireup.h

```diff
@@ -13,10 +13,14 @@
 #include <uct/api/uct.h>
 #include <ucs/arch/bitops.h>
 
+/* Struct used by select criteria to allow flexibility on required flags:
+ * - mandatory: All flags specified by this field must be set.
+ * - optional:  In addition to all mandatory flags, at least one of the flags
```

Comment:
fields documentation should be near the fields
see https://github.com/openucx/ucx/blob/master/src/ucp/rndv/proto_rndv.h#L29 as example